src/azure-cli-core/azure/cli/core/aaz/_field_value.py

```diff
@@ -444,3 +445,93 @@ def to_serialized_data(self, processor=None, keep_undefined_in_list=False,  # py
         if processor:
             result = processor(self._schema, result)
         return result
+
+
+class AAZIdentityObject(AAZObject):  # pylint: disable=too-few-public-methods
+    def to_serialized_data(self, processor=None, **kwargs):
+        calculate_data = dict()
+        if self._data == AAZUndefined:
+            result = AAZUndefined
+
+        elif self._data is None:
+            result = None
+
+        else:
+            result = {}
+            schema = self._schema
+
+            for name, field_schema in schema._fields.items():
+                if name in self._data:
+                    v = self[name].to_serialized_data(processor=processor, **kwargs)
+                    if v == AAZUndefined:
+                        continue
+
+                    if field_schema._serialized_name:
+                        name = field_schema._serialized_name
+
+                    if name in {"userAssigned", "systemAssigned"}:
+                        calculate_data[name] = v
+                        calculate_data["action"] = field_schema._flags.get("action", None)  # no action in GET operation
```

Comment:
By moving the action in IdentityObject, you can use the following way to get the action for calculation:
```
```suggestion
                        calculate_data["action"] = self._schema._flags.get("action", None) # no action in GET operation
```